### PR TITLE
docgen updates

### PR DESCRIPTION
**Pull request by xPaw**
This PR replaces crappy search with a proper one that filters the sidebar.

For improved performance, every `name` column should have an index added.

Other changes:
- Added a check to see if include exists before trying to find objects in it and renders 404 page if not found
- Breadcrumb improvements so that it actually says where you are now
- Added a panel saying there are no types/functions if include file exists but there is nothing to display
- Added a label with return type in return value section
- Made custom types clickable
- Added some poor type highlighting in syntax display
- A bunch of minor style related fixes

### CODE REVIEW

**Comment by xPaw** [comment]
Also did a couple of fixes with `active` functions highlight in sidebar. @dvander if you'd like, I can squash all my changes.

File: exp/docgen/www/index.php [hunk 1]
```diff
@@ -6,17 +6,35 @@
     
     $Path = isset( $_SERVER[ 'QUERY_STRING' ] ) ? trim( $_SERVER[ 'QUERY_STRING' ], '/' ) : '';
     $RenderLayout = !isset( $_SERVER[ 'HTTP_X_PJAX' ] ) || $_SERVER[ 'HTTP_X_PJAX' ] !== 'true';
-    
-    if (substr($Path, 0, 8) === '__search')
-        $RenderLayout = false;
-    
+
+    // Stupid bugfix when clicking on homepage
+    if (substr($Path, 0, 6) === '_pjax=') {
```
Comment (dvander) [inline exp/docgen/www/index.php]: What exactly is this?

Comment (xPaw) [inline exp/docgen/www/index.php]: Basically, mod_rewrite doesn't pass in path correctly when you try to load homepage using pjax (`/new-api/?_pjax=testest`) resulting in a 404 error. There's most likely a cleaner way of fixing this via `.htaccess`, but I couldn't figure it out.

File: exp/docgen/www/template/function.php [hunk 2]
```diff
@@ -22,41 +45,94 @@
...
-
-<h4 class="sub-header2">Description</h4>
-<?php RenderDescription($PageFunction['brief']); ?>
+<?php if (empty($PageFunction['brief'])) { ?>
+  <p class="text-muted">This function has no description.</p>
+<?php } else { RenderDescription($PageFunction['brief']); } ?>
+
+<?php
+if (isset($Data['return'])) {
+    $Type = NormalizeType($Data['return']['type']);
```
Comment (dvander) [inline exp/docgen/www/template/function.php]: This is something that should be cross-indexed in the database itself, but that can wait for an eventual docgen rewrite.

**Pull request merged**